Analytics/Classes/Internal/SEGSegmentIntegration.m

```diff
@@ -466,10 +466,12 @@ - (void)flushWithMaxSize:(NSUInteger)maxBatchSize
     [self dispatchBackground:^{
         if ([self.queue count] == 0) {
             SEGLog(@"%@ No queued API calls to flush.", self);
+            [self endBackgroundTask];
             return;
         }
         if (self.batchRequest != nil) {
             SEGLog(@"%@ API request already in progress, not flushing again.", self);
+            [self endBackgroundTask];
```

Comment:
There is already a flush in progress (which will take care of ending the background task), so I don't think we should end it here.